core/src/test/java/com/netease/arctic/catalog/IcebergCatalogTest.java

```diff
@@ -34,16 +41,73 @@ public IcebergCatalogTest() {
 
   @Test
   public void testLoadIcebergTable() {
-    ArcticCatalog icebergCatalog = getCatalog();
-    icebergCatalog.createDatabase("db2");
+    getCatalog().createDatabase(TableTestHelpers.TEST_DB_NAME);
     Catalog nativeIcebergCatalog = getIcebergCatalog();
-    nativeIcebergCatalog.createTable(TableIdentifier.of("db2", "tb1"), TableTestHelpers.TABLE_SCHEMA);
-    ArcticTable table = icebergCatalog.loadTable(
-        com.netease.arctic.table.TableIdentifier.of(TEST_CATALOG_NAME, "db2", "tb1"));
+    nativeIcebergCatalog.createTable(
+        TableIdentifier.of(TableTestHelpers.TEST_DB_NAME, TableTestHelpers.TEST_TABLE_NAME),
+        TableTestHelpers.TABLE_SCHEMA);
+    ArcticTable table = getCatalog().loadTable(TableTestHelpers.TEST_TABLE_ID);
     Assert.assertTrue(table instanceof BasicIcebergCatalog.BasicIcebergTable);
     Assert.assertEquals(true, table.isUnkeyedTable());
     Assert.assertEquals(TableTestHelpers.TABLE_SCHEMA.asStruct(), table.schema().asStruct());
-    nativeIcebergCatalog.dropTable(TableIdentifier.of("db2", "tb1"), true);
-    icebergCatalog.dropDatabase("db2");
+  }
+
+  @Test
+  public void testCreateTableWithCatalogTableProperties() throws TException {
+    CatalogMeta testCatalogMeta = TEST_AMS.getAmsHandler().getCatalog(TEST_CATALOG_NAME);
+    TEST_AMS.getAmsHandler().updateMeta(testCatalogMeta,
+        CatalogMetaProperties.TABLE_PROPERTIES_PREFIX + TableProperties.ENABLE_SELF_OPTIMIZING,
+        "false");
+    TEST_AMS.getAmsHandler().updateMeta(testCatalogMeta,
+        CatalogMetaProperties.TABLE_PROPERTIES_PREFIX + TableProperties.ENABLE_TABLE_TRASH,
+        "true");
+    getCatalog().createDatabase(TableTestHelpers.TEST_DB_NAME);
+    Catalog nativeIcebergCatalog = getIcebergCatalog();
+    nativeIcebergCatalog.createTable(
+        TableIdentifier.of(TableTestHelpers.TEST_DB_NAME, TableTestHelpers.TEST_TABLE_NAME),
+        TableTestHelpers.TABLE_SCHEMA);
+    ArcticTable createTable = getCatalog().loadTable(TableTestHelpers.TEST_TABLE_ID);
+    Assert.assertEquals(false, PropertyUtil.propertyAsBoolean(createTable.properties(),
+        TableProperties.ENABLE_SELF_OPTIMIZING, TableProperties.ENABLE_SELF_OPTIMIZING_DEFAULT));
+    Assert.assertEquals(true, PropertyUtil.propertyAsBoolean(createTable.properties(),
+        TableProperties.ENABLE_TABLE_TRASH, TableProperties.ENABLE_TABLE_TRASH_DEFAULT));
+    Assert.assertFalse(createTable.io() instanceof RecoverableArcticFileIO);
+  }
+
+  @Test
+  public void testLoadTableWithNewCatalogProperties() throws TException {
```

Comment:
If this test case is suitable for all the catalog types, I'd recommend to move it to `BaseCatalogTest`.

Reply:
I moved both `testLoadTableWithNewCatalogProperties` and `testCreateTableWithCatalogTableProperties` to `BaseCatalogTest`.